### ISSUE TITLE
TypeError: the JSON object must be str, not 'bytes'

### ISSUE THREAD

**Issue by slowkow** [jupyterlab/jupyterlab-toc#26]
I successfully installed and use jupyterlab-toc:

```
jupyter labextension list
JupyterLab v0.32.1
Known labextensions:
   app dir: /data/srlab/slowikow/src/miniconda2/envs/py3/share/jupyter/lab
jupyterlab-toc
        jupyterlab-toc v0.2.1  enabled  OK
```

Today, I decided to try to upgrade to v0.3.1, but I get an error:

```
jupyter labextension install @jupyterlab/toc
> /data/srlab/slowikow/src/miniconda2/envs/py3/bin/npm pack @jupyterlab/toc
jupyterlab-toc-0.3.1.tgz

Errored, use --debug for full output:
TypeError: the JSON object must be str, not 'bytes'
```

It seems like [an easy bug to fix][1], but I don't know which file has the buggy code.

[1]: https://stackoverflow.com/questions/42683478/typeerror-the-json-object-must-be-str-not-bytes/42683509

I hope you can help!

Thanks for the great extension 👍 

Here are more details:

<details>
<summary>jupyter labextension install --debug @jupyterlab/toc</summary>

<pre>
jupyter labextension install --debug @jupyterlab/toc
Searching ['/PHShome/ks38', '/PHShome/ks38/.jupyter', '/data/srlab/slowikow/src/miniconda2/envs/py3/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
Looking for jupyter_config in /etc/jupyter
Looking for jupyter_config in /usr/local/etc/jupyter
Looking for jupyter_config in /data/srlab/slowikow/src/miniconda2/envs/py3/etc/jupyter
Looking for jupyter_config in /PHShome/ks38/.jupyter
Looking for jupyter_config in /PHShome/ks38
> /data/srlab/slowikow/src/miniconda2/envs/py3/bin/npm pack @jupyterlab/toc
jupyterlab-toc-0.3.1.tgz

   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.6.0 → 6.2.0    │
   │     Run npm i -g npm to update      │
   │                                     │
   ╰─────────────────────────────────────╯

Fetching URL: https://registry.npmjs.org/@jupyterlab%2Ftoc
Traceback (most recent call last):

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/labextensions.py", line 71, in start
    ans = self.run_task()

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/labextensions.py", line 100, in run_task
    for arg in self.extra_args

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/labextensions.py", line 100, in <listcomp>
    for arg in self.extra_args

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py", line 227, in install_extension
    return handler.install_extension(extension)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py", line 366, in install_extension
    info = self._install_extension(extension, tempdir)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py", line 1138, in _install_extension
    version = self._latest_compatible_package_version(name)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py", line 1202, in _latest_compatible_package_version
    metadata = _fetch_package_metadata(self.registry, name, self.logger)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py", line 1588, in _fetch_package_metadata
    return json.load(response)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/json/__init__.py", line 268, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)

  File "/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))

TypeError: the JSON object must be str, not 'bytes'

Errored, use --debug for full output:
TypeError: the JSON object must be str, not 'bytes'

</pre>
</details>

**Comment by ian-r-rose** [comment]
What version of JupyterLab are you using? You may have caught us in the middle of the release cycle: `v0.3.1` is targeting the new release of JupyterLab (v0.33, still in RC).

**Comment by slowkow** [comment]
Yes, I think you're right... but there may also be a bug.

I didn't check my version before upgrading, but I can verify that I got this message with the old version (unknown) of jupyterlab and with the new version (0.32.1):

```
Errored, use --debug for full output:
ValueError: No version of @jupyterlab/toc could be found that is
 compatible with the current version of JupyterLab. However, it
 seems to support a new version of JupyterLab.
 Consider upgrading JupyterLab.

Conflicting Dependencies:
JupyterLab              Extension        Package
>=0.16.3 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/rendermime
>=0.16.3 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/notebook
>=0.16.4 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/apputils
>=0.16.3 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/fileeditor
>=1.1.3 <2.0.0          >=2.0.0 <3.0.0   @jupyterlab/coreutils
>=0.16.3 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/application
>=0.16.3 <0.17.0        >=0.17.0 <0.18.0 @jupyterlab/docmanager
```

To get this error, first I had to fix a bug in this file:

```
/data/srlab/slowikow/src/miniconda2/envs/py3/lib/python3.5/site-packages/jupyterlab/commands.py
```

I changed line 1588 in `_fetch_package_metadata()` from this:

```python
return json.load(response)
```

to this:

```python
return json.loads(response.read().decode('utf-8'))
```

This got me past the issue with `TypeError: the JSON ...`, but now I get the new issue complaining that my jupyterlab version is too old.

**Comment by ian-r-rose** [comment]
Interesting. What version of Python are you using? I may have seen this before with 3.5...

**Comment by slowkow** [comment]
Yep, you are right:

```
python --version
Python 3.5.2 :: Continuum Analytics, Inc.
```

**Comment by ian-r-rose** [comment]
Great, thanks for the report @slowkow! This fix will need to go into the core JupyterLab python package.

**Comment by ian-r-rose** [comment]
Just opened jupyterlab/jupyterlab#4962 with this fix. Thanks again @slowkow

**Comment by ian-r-rose** [comment]
Fixed by jupyterlab/jupyterlab#4692
